src/actions.ml

```diff
@@ -463,41 +463,68 @@ let create_pipeline_summary ?summary_top pipeline_info pipeline_url =
   |> (match summary_top with Some text -> List.cons text | None -> Fn.id)
   |> String.concat ~sep:"\n\n"
 
-let run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo ~docker_image
-    ~target ~opam_switch ~failing_urls ~passing_urls ~base ~head =
-  git_run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo
-    ~docker_image ~target ~opam_switch ~failing_urls ~passing_urls ~base ~head
-  >>= function
-  | Ok () ->
-      GitHub_mutations.post_comment ~id:comment_thread_id
-        ~message:
-          (f
-             "Hey, I have detected that there was a failure in project %s (for \
-              commit %s) without any failure in the test-suite.\n\
-              I checked that the corresponding job for the base commit %s \
-              succeeded. Now, I'm trying to extract a minimal test case from \
-              this so that it can be added to the test-suite. I'll come back \
-              to you with the results once it's done."
-             target head base)
-        ~bot_info
-      >>= GitHub_mutations.report_on_posting_comment
-  | Error f ->
-      Lwt_io.printf "Error: %s\n" f
+type ci_minimization_info =
+  { target: string
+  ; full_target: string
+  ; docker_image: string
+  ; opam_switch: string
+  ; failing_urls: string
+  ; passing_urls: string }
+
+let run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo ~base ~head
+    ~ci_minimization_infos =
+  (* XXX Will parallel map cause interference with git locks? *)
+  Lwt_list.map_p
+    (fun {target; opam_switch; failing_urls; passing_urls; docker_image} ->
+      git_run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo
+        ~docker_image ~target ~opam_switch ~failing_urls ~passing_urls ~base
+        ~head
+      >>= fun result -> Lwt.return (target, result))
+    ci_minimization_infos
+  >>= fun results ->
+  results
+  |> List.partition_map ~f:(function
+       | target, Ok () ->
+           Either.First target
+       | target, Error f ->
+           Either.Second (target, f))
```

Comment:
Why an `Either` here? This could also have been:
```suggestion
       | target, Ok () ->
           Ok target
       | target, Error f ->
           Error (target, f))
```

Reply:
Is there a version of `partition_map` for `result`?  If not, then what you suggest doesn't typecheck, and I want to have both lists separately

Reply:
Hum, OK, that's not likely.